js/ui/sessionMode.js

```diff
@@ -153,6 +177,11 @@ const SessionMode = new Lang.Class({
             let primary = modes[global.session_mode] &&
                           modes[global.session_mode].isPrimary;
             let mode = primary ? global.session_mode : 'user';
+
+            if (mode == 'user' &&
+                global.settings.get_boolean('enable-mission-game'))
+                mode = 'user-mission';
```

Comment:
Ah interesting - that's a handy way to enable and disable this stuff without too much invasive change. Though that said, I think the effect of this change will be that the mission game mode will only be enabled across login/logout (and we likely won't be able to apply the setting immediately, for instance, if the user decides they want to turn it off for a while).

We might be okay with that for now though, so if that's good, then this is fine by me to merge.